Mage.Tests/src/test/java/org/mage/test/cards/single/cmm/DemonOfFatesDesignTest.java

```diff
@@ -0,0 +1,229 @@
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        assertPermanentCount(playerA, "Glorious Anthem", 1);
+        assertPermanentCount(playerA, "Absolute Law", 1);
+        assertLife(playerA, 20 - 3 - 2);
+    }
+
+    @Test
+    public void UnsubstantiateTest() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, demon, 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 2);
+
+        // Instant {U}{U}
+        // Return target spell or creature to its owner’s hand.
+        addCard(Zone.HAND, playerA, "Unsubstantiate");
+        addCard(Zone.HAND, playerA, "Glorious Anthem"); // Enchantment {1}{W}{W}
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Glorious Anthem");
+        setChoice(playerA, true); // yes to alt cast
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Unsubstantiate", "Glorious Anthem");
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        assertHandCount(playerA, "Glorious Anthem", 1);
+        assertLife(playerA, 20 - 3);
+
+        // Did not keep the alt cost from the first cast
+        castSpell(1, PhaseStep.POSTCOMBAT_MAIN, playerA, "Glorious Anthem");
+
+        boolean hadError = false;
+        try {
+            setStopAt(1, PhaseStep.END_TURN);
+            execute();
+        } catch (AssertionError e) {
+            hadError = true;
+            assert e.getMessage().equals("Can't find ability to activate command: Cast Glorious Anthem");
+        } finally {
+            assert hadError;
+        }
+    }
+
+    @Test
+    public void DoubleDemonDoubleCast() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, demon, 2);
+
+        addCard(Zone.HAND, playerA, "Glorious Anthem"); // Enchantment {1}{W}{W}
+        addCard(Zone.HAND, playerA, "Absolute Law"); // Enchantment {1}{W}
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Glorious Anthem", true);
+        setChoice(playerA, false); // no to the first one
+        setChoice(playerA, true); // yes to second one
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Absolute Law");
+        setChoice(playerA, true); // yes to first one
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        assertPermanentCount(playerA, "Glorious Anthem", 1);
+        assertPermanentCount(playerA, "Absolute Law", 1);
+        assertLife(playerA, 20 - 3 - 2);
+    }
+
+    //118.9a Only one alternative cost can be applied to any one spell as it’s being cast.
+    @Test
+    public void DoubleDemon() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, demon, 2);
+
+        addCard(Zone.HAND, playerA, "Glorious Anthem"); // Enchantment {1}{W}{W}
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Glorious Anthem");
+        setChoice(playerA, true); // yes to alt cast of first demon
+        setChoice(playerA, true); // yes to alt cast of second demon ???
```

Comment:
Looks like a bug during AlternativeCost choice/payment.
There should be no choice after the first being "yes cast with this alternative cost"